src/sunmemory/hip/sundials_hip_memory.hip.cpp

```diff
@@ -382,6 +395,6 @@ int SUNMemoryHelper_GetAllocStats_Hip(SUNMemoryHelper helper,
     *bytes_allocated      = SUNHELPER_CONTENT(helper)->bytes_allocated_uvm;
     *bytes_high_watermark = SUNHELPER_CONTENT(helper)->bytes_high_watermark_uvm;
   }
-  else { return -1; }
-  return 0;
+  else { return SUN_ERR_CORRUPT; }
```

Comment:
You'd used `SUN_ERR_OUTOFRANGE` in the CUDA version, but in both I think it should be:
```suggestion
  else { return SUN_ERR_INCOMPATIBLE; }
```

Reply:
Indeed `CORRUPT` is the wrong code to use for this case. However, in another PR we used `OUTOFRANGE` for a similar use (a value outside the range of an `enum`-like value), so Ill go with that.   To me `INCOMPATIBLE` implies the value is valid but not usable with the object or its current configuration while `OUTOFRANGE` indicates its a value that is outside the valid set.